iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/scoped_static.inl

```diff
@@ -27,7 +27,7 @@ template <typename T, typename... CTorArgs>
 inline ScopeGuard makeScopedStatic(T& memory, CTorArgs&&... ctorArgs) noexcept
 {
     memory.emplace(std::forward<CTorArgs>(ctorArgs)...);
-    return ScopeGuard([&memory] { memory.reset(); });
+    return ScopeGuard([&memory]() noexcept { memory.reset(); });
```

Comment:
Well, this actually depends on `T` :thinking:

Reply:
@FerdinandSpitzschnueffler What do you think about implementing `std::is_nothrow_invocable` and using this type trait here to narrow down the usage only to types which `reset` method can't throw?

Reply:
@FerdinandSpitzschnueffler No, then this lambda would be always `noexcept`, Axivion would not complain and we would verify that `T::reset()` is `noexcept` too.

Reply:
Turns out, the `noexcept` wasn't needed after all as it was not related to an AUTOSAR rule but to `FaultDetection-NoexceptViolations`. I removed it.